apis/iap/v1alpha1/iapsettings_identity.go

```diff
@@ -198,42 +200,37 @@ func (p ComputeServiceParent) buildIAPSettingsID(ctx context.Context, reader cli
 // AppEngineParent represents project-wide App Engine service settings
 type AppEngineParent struct {
 	ProjectRef     *refsv1beta1.ProjectRef
-	ApplicationRef *refsv1beta1.AppEngineApplicationRef
-	ServiceRef     *refsv1beta1.AppEngineServiceRef
-	VersionRef     *refsv1beta1.AppEngineVersionRef
+	ApplicationRef *appenginev1beta1.AppEngineApplicationRef
+	ServiceRef     *appenginev1beta1.AppEngineServiceRef
+	VersionRef     *appenginev1beta1.AppEngineVersionRef
 }
 
 func (p AppEngineParent) buildIAPSettingsID(ctx context.Context, reader client.Reader, namespace string) (string, error) {
-	project, err := refsv1beta1.ResolveProject(ctx, reader, namespace, p.ProjectRef)
-	if err != nil {
-		return "", err
+	if p.ApplicationRef == nil {
+		return "", fmt.Errorf("applicationRef is required")
 	}
 
-	appID, err := refsv1beta1.ResolveAppEngineApplicationID(ctx, reader, namespace, p.ApplicationRef)
+	applicationExternal, err := p.ApplicationRef.NormalizedExternal(ctx, reader, namespace)
 	if err != nil {
 		return "", err
 	}
 
 	if p.ServiceRef != nil {
-		serviceID, err := refsv1beta1.ResolveAppEngineServiceID(ctx, reader, namespace, p.ServiceRef)
+		serviceExternal, err := p.ServiceRef.NormalizedExternal(ctx, reader, namespace)
 		if err != nil {
 			return "", err
 		}
-
-		if p.VersionRef != nil {
-			versionID, err := refsv1beta1.ResolveAppEngineVersionID(ctx, reader, namespace, p.VersionRef)
-			if err != nil {
-				return "", err
-			}
-			return fmt.Sprintf("projects/%s/iap_web/appengine-%s/services/%s/versions/%s",
-				project.ProjectID, appID, serviceID, versionID), nil
+		return serviceExternal, nil
+	}
+	if p.VersionRef != nil {
```

Comment:
It appears that we are changing the logic here? I think we are missing test cases for these fields and therefore the logic change was undetected.